### DISCUSSION TITLE
Release cycles

### DISCUSSION THREAD

**Discussion by AlexanderRavenheart** [FreeTubeApp/FreeTube#3768] [General]
Thank you for all the hard work put into this and associated projects.
I have noticed that the development branch is quite active.

Since a Youtube client has to be always aligned with the latest changes in the layout and API changes of the source platform, any older releases quickly become unusable.

If manually creating new releases/tags is too much of a chore, please define a Github Action to generate a new beta/dev version every month, so that anyone interested in this app can use it without relying on having to manually download a Nightly build.

At the very least, the monthly builds should be pushed to Flathub and AUR, since those repositories were designed for hosting the very latest updates.

**Comment by absidue** [comment]
It's not about being too much work to create releases, it's that we don't want to release stable builds that are worse than the existing stable release. If we had done an automated release every month since 0.18.0 was released, they would have had a half finished local API migration (took 4 months), much buggier local API (took another few months to squash a lot of bugs that were introduced in the migration), significantly worse playback issues for months than on 0.18.0, we only finally figured out the cause of that a few days ago. There is a reason the nightly builds are called nightly.

We still aren't quite ready to do the next release, because we haven't got a big feature completed/merged in yet, that everyone requested for the next release.

Additionally the FreeTube AUR releases are unofficial and not maintained by us, if you want buggy unstable nightly releases through AUR you need to contact the maintainers over there.

**Comment by AlexanderRavenheart** [comment]
Do you consider 0.18.0 a stable release?
- It is marked as a beta (pre-release) version, just like most of the other releases of this project
That is normal, given that the project is pretty much in the early stages and it is quickly iterating with breaking changes in almost every release until it reaches what would be considered a 1.0 feature complete  and stable version.
- Have you tried using that version recently? 
For me it is mostly unusable due to very slow buffering, video playback freezes and sometimes completely refusing to load the rest of the video (constant loading icon being displayed on top of the video frame).

While I agree it is not ideal to deliver incomplete/in-progress builds, I believe it is something for the user to decide, since all package management software can be configured to only install stable versions or allow installation of dev/beta/pre-release versions.

I think that for a fast-evolving project like this one it is better to have more builds available in a timely fashion, that users can test and provide feedback on, rather than leaving them waiting for many months for critical fixes because you are delaying the next release until a specific feature is completed.

Perhaps make a 0.18.1 release with just critical bug fixes, leaving any big refactoring and other features being worked on for a 0.19.0 release. The way this could be handled in git would be to have a separate branch for each minor version bump (release_0.18 for example), where bug fixes could be merged to and from where new point releases could be made for that specific version.

I am interested in hearing what other users and contributors have to say about this topic and what would be a good approach for everyone involved.
I ended up downloading one of the Nightly AppImage builds, and it has been working fine so far.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
IMO users cant decide anything about release cycles because they don't understand what is going on behind the scenes.

FT is marked as pre release and beta because its not feature complete. If we match feature parity with YT then it will go to v1.0.0 but that doesn't mean we will be stable as YT is always changing things. That is just the way it is and always will be.

Creating a v0.18.1, that doesnt make any sense right now because we almost ready for RC.

**Comment by VictorVow** [comment]
Please can we at least have Flathub-Beta builds then? The current release is in a poor state now thanks to server-side changes by YouTube (e.g. #3773) that have made playback much worse but is fixed on nightly. Also the "sort by most popular" on channel pages doesn't work on the latest release, and there are a bunch of nice features in the current nightly like Shorts tab and Live.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I'm sorry but i'm going to close this discussion. We understand that a new release is needed and we are working hard on delivering it. Like said earlier this release took very long because of the migration and all the bugs it brought with it.

All the things being asked for in this discussion are just the result of the long wait and to prevent another long period of waiting in the future.

We strive to put out releases faster in the future and improve our workflows but asking for all these things will only stall our dev work.

Right now our focus is on delivering the RC.
